e2e/api/api_test.go

```diff
@@ -332,21 +389,113 @@ var _ = Describe("API", func() {
 
 		Describe("UpdateServer", func() {
 
-			It("should return codes.NotFound if doesn't exist", func() {
+			// test all invalid / missing fields (missing key, missing service, missing server)
+			DescribeTable("should return codes.NotFound if doesn't exist", func(update *types.RealServer) {
 				server := &types.RealServer{
-					ServiceID: "service1",
+					ServiceID: service.Id,
 					Key:       validServerKey,
 					Config:    validServerConfig,
 				}
-				_, err := client.UpdateServer(ctx, server)
+				_, err := client.CreateServer(ctx, server)
+				Expect(err).ToNot(HaveOccurred())
+
+				_, err = client.UpdateServer(ctx, update)
 				status, ok := status.FromError(err)
 
 				Expect(ok).To(BeTrue(), "got grpc status error")
 				if ok {
```

Comment:
wouldn't the test fail on the previous expectation making this check unecesary?
I saw another one somewhere up.

Reply:
It doesn't stop the test from executing, the failures are cumulative. If I don't check ok it will panic.

Reply:
Maybe it is a behaviour of the DescribeTable? any other execution will stop the test scenario.

Reply:
Yep you're right, this seems specific to ginkgo. Confusing, since golang's testing framework will carry on when you call `t.Fail()`.